### DISCUSSION TITLE
Way to see Diarization Speaker-Probabilities by word?

### DISCUSSION THREAD

**Discussion by gatestonjohns** [pyannote/pyannote-audio#1632] [Q&A]
Essentially, I am simply looking for a way to see the diarization probabilities for each speaker on any given word (or segment, if word is not possible). I have tracked down the functionality to be within the apply function of the speaker_diarization pipeline, but it is difficult to understand what is going on exactly without serious examination and time commitment. Any help would be appreciated, thanks!

**Comment by hbredin** [comment]
Related to #1636. Duplicating my answer here:

Out of the box, there is no such thing available in pyannote.

This is a tricky problem because the underlying algorithm is a combination of multiple parts, each of them having their own uncertainties, and some of them (e.g. the clustering part) don't even provide such a thing.

**Reply by gatestonjohns** [reply]
Is there any way we may be able to **_calculate a rough estimate of the diarization confidence using the soft_clusters matrix returned by the Agglomerative Clustering algorithm_**? I do not need an exactly correct probability matrix, just some type of indication when words are not super confidently diarized.

**Comment by qalabeabbas49** [comment]
Hi, any progress on this? I am also looking for something similar.

Thanks
